src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -33,13 +33,18 @@ public DeleteCommand parse(String args) throws ParseException {
             }
         }
 
+        String preamble = argMultimap.getPreamble();
+        if (preamble.matches("^[0-9]+$")) {
+            throw new ParseException(DeleteCommand.MESSAGE_MULTIPLE_WAYS_FORBIDDEN);
+        }
+
         String str = argMultimap.getValue(PREFIX_NAME).get();
         try {
             Name name = ParserUtil.parseName(str);
             return new DeleteCommand(name);
         } catch (Exception ex) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), ex);
+            throw new ParseException(MESSAGE_CONTACT_NOT_IN_ADDRESS_BOOK);
```

Comment:
This is not the correct message. Original message is responding to "delete n/James&", where name specified is invalid.

Message_contact_not_in_address_book is only happens during execution when targetIndex is not found.